### PR TITLE
INFRA-1105 - Rename the image to jenkins/ssh-agent, stop building generic Windows tags

### PR DESCRIPTION
**Pull request by oleg-nenashev**
Updates the documentation before the image update. https://issues.jenkins-ci.org/browse/INFRA-1105

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
@jenkinsci/team-docker-packaging Should we keep the image name as `jenkins/ssh-build-agent`. Or should we rename to something like `ssh-outbound-agent` or so?

**Comment by oleg-nenashev** [comment]
UPD for @slide @timja . Windows images here deploy to `jenkins/ssh-agent`, so t is safe to proceed as is. Otherwise we may get into collisions between tags

**Comment by jglick** [comment]
`jenkins/ssh-agent` would be the natural name to me (matches `jenkins/agent` and `jenkins/inbound-agent`; we do not generally include the redundant _build_ terminology).

File: README.md [hunk 1]
```diff
@@ -13,14 +14,14 @@ See [Jenkins Distributed builds](https://wiki.jenkins-ci.org/display/JENKINS/Dis
 To run a Docker container
 
 ```bash
-docker run jenkins/ssh-slave "<public key>"
+docker run jenkins/ssh-build-agent "<public key>"
```
Comment (slide) [inline README.md:17]: should this be just `ssh-agent`?

Comment (oleg-nenashev) [reply README.md:17]:
```suggestion
docker run jenkins/ssh-agent "<public key>"
```

Yes, leftover

**Comment by slide** [comment]
:shipit:

**Comment by oleg-nenashev** [comment]
FTR I summarized the voting status in https://groups.google.com/d/msg/jenkinsci-dev/oxD-Hd_7l9k/WTnoTkyDAQAJ . Will go ahead and merge it, but we have a few extra days to revisit the decision

**Pull request merged**